lib/redcord/serializer.rb

```diff
@@ -31,7 +31,7 @@ module ClassMethods
     sig { params(attribute: Symbol, val: T.untyped).returns(T.untyped) }
     def encode_attr_value(attribute, val)
       if !val.blank? && TIME_TYPES.include?(props[attribute][:type])
-        val = val.to_f
+        val = val.to_i * 1_000_000_000 + val.nsec
```

Comment:
What if `val.to_i` is negative? Do we need to raise an error or support such case?

Reply:
hmm I'm not sure when this gonna happen when it's a TIME_TYPES  but happy to add a validation

Reply:
It can be when time is before year 1970.